synapse/config/workers.py

```diff
@@ -26,6 +27,16 @@
 )
 from .server import ListenerConfig, parse_listener_def
 
+logger = logging.Logger(__name__)
+
+# an object to pass to the "default" parameter, with which we can
+# `is` against to be sure we have an undefined config option.
+UNDEFINED = object()
```

Comment:
convention is to call things like this `SENTINEL` rather than `UNDEFINED`, though tbh I think it would be better to avoid the need for it with an explicit test like:

```
if 'update_user_directory' in config:
```

Reply:
Ah, wasn't aware of that trick, will apply it. :+1: